### PR TITLE
[MERGE][IMP] microsoft_outlook, fetchmail_outlook: improve usability, lessen module complexity

### PR DESCRIPTION
**Pull request by std-odoo**
Purpose
=======

Improve the usability of the outlook modules. Lessen module complexity.

Specifications
==============

Remove the checkbox "Use Outlook" and instead use the
smtp_authentication and the server_type, to be consistent with Gmail.

Hide the password field for Outlook / Gmail mail servers.

Add constraints on the outgoing mail server to force the user to use
the right configuration (e.g. the from_filter, so the sending do not
fail).

Merge fetchmail_{google/microsoft} into google_gmail / microsoft_outlook?
Most of the time, both module will be installed at the same time, or
none. For that reason, we want to merge both modules, for both
providers, to lessen complexity.

Task-2811567

### CODE REVIEW

File: addons/microsoft_outlook/views/ir_mail_server_views.xml [hunk 1]
```diff
@@ -5,34 +5,30 @@
...
             <field name="smtp_user" position="after">
                 <field name="is_microsoft_outlook_configured" invisible="1"/>
                 <field name="microsoft_outlook_refresh_token" invisible="1"/>
                 <field name="microsoft_outlook_access_token" invisible="1"/>
                 <field name="microsoft_outlook_access_token_expiration" invisible="1"/>
-                <div></div>
-                <div attrs="{'invisible': [('use_microsoft_outlook_service', '=', False)]}">
-                    <span attrs="{'invisible': ['|', ('use_microsoft_outlook_service', '=', False), ('microsoft_outlook_refresh_token', '=', False)]}"
+                <div attrs="{'invisible': [('smtp_authentication', '!=', 'outlook')]}"
+                    class="d-flex flex-row align-items-center">
```
Comment (tde-banana-odoo) [inline addons/microsoft_outlook/views/ir_mail_server_views.xml:14]: That's quite uncommon to use custom styling in form views, can't it be done using standard form rendering ?

**Review by tde-banana-odoo — COMMENTED** [review]
Validation ! LGTM, some small comments :)

File: addons/microsoft_outlook/views/ir_mail_server_views.xml (hunk 1 above)
Comment (std-odoo) [reply addons/microsoft_outlook/views/ir_mail_server_views.xml:14]: we can, but it won't be 100% centered with standard "row" class

File: addons/microsoft_outlook/models/microsoft_outlook_mixin.py [hunk 2]
```diff
@@ -35,22 +34,20 @@ class MicrosoftOutlookMixin(models.AbstractModel):
     microsoft_outlook_uri = fields.Char(compute='_compute_outlook_uri', string='Authentication URI',
         help='The URL to generate the authorization code from Outlook', groups='base.group_system')
 
-    @api.depends('use_microsoft_outlook_service')
     def _compute_is_microsoft_outlook_configured(self):
```
Comment (tde-banana-odoo) [inline addons/microsoft_outlook/models/microsoft_outlook_mixin.py:37]: Would keep at least one trigger (even dummy)

File: addons/microsoft_outlook/models/microsoft_outlook_mixin.py [hunk 3]
```diff
@@ -35,22 +34,20 @@ class MicrosoftOutlookMixin(models.AbstractModel):
...
         Config = self.env['ir.config_parameter'].sudo()
         microsoft_outlook_client_id = Config.get_param('microsoft_outlook_client_id')
         microsoft_outlook_client_secret = Config.get_param('microsoft_outlook_client_secret')
         self.is_microsoft_outlook_configured = microsoft_outlook_client_id and microsoft_outlook_client_secret
 
-    @api.depends('use_microsoft_outlook_service')
     def _compute_outlook_uri(self):
```
Comment (tde-banana-odoo) [inline addons/microsoft_outlook/models/microsoft_outlook_mixin.py:43]: Would keep at least one trigger (even dummy)

File: addons/microsoft_outlook/models/ir_mail_server.py [hunk 4]
```diff
@@ -15,10 +15,32 @@ class IrMailServer(models.Model):
...
+            'for your current session. \n'
+            'If you plan to use only this account, you should set "mail.default.from" '
+            'in the system parameter view to your Outlook email address.')
+        super(IrMailServer, self - outlook_servers)._compute_smtp_authentication_info()
+
+    @api.constrains('smtp_authentication', 'smtp_pass', 'smtp_encryption', 'from_filter', 'smtp_user')
     def _check_use_microsoft_outlook_service(self):
         for server in self:
-            if not server.use_microsoft_outlook_service:
+            if server.smtp_authentication != 'outlook':
```
Comment (tde-banana-odoo) [inline addons/microsoft_outlook/models/ir_mail_server.py:43]: Could be inlined in the loop header with a filtered

File: addons/google_gmail/__manifest__.py [hunk 5]
```diff
@@ -7,10 +7,11 @@
     "category": "Hidden",
     "description": "Gmail support for incoming / outgoing mail servers",
     "depends": [
+        "fetchmail",
         "mail",
-        "google_account",
```
Comment (tde-banana-odoo) [inline addons/google_gmail/__manifest__.py:11]: Don't we have to keep google account anymore ?

**Review by tde-banana-odoo — COMMENTED** [review]
Some comments while re-reading :)

**Review by std-odoo — COMMENTED** [review]
@tde-banana-odoo Hi :) 

Changes done and answered :)

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge

**Comment by adr-odoo** [comment]
It seems that this PR has been merged after the bootstrap migration but there is still BS4 classes.
Can I ask you to fix this ASAP please.
https://runbot.odoo.com/runbot/batch/810834/build/18463658

**Comment by std-odoo** [comment]
@adr-odoo Hi :) 

Can you review / R+ ? https://github.com/odoo/odoo/pull/98277/files

**Pull request closed without merging**